src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java

```diff
@@ -151,17 +152,16 @@ public Case findOne(String caseId) {
 
     @Override
     public List<Case> findAllById(List<String> ids) {
-        List<Case> page = new LinkedList<>();
-        ids.forEach(id -> {
-            Optional<Case> useCase = repository.findById(id);
-            useCase.ifPresent(page::add);
-        });
-        if (page.size() > 0) {
-            page.forEach(c -> c.setPetriNet(petriNetService.get(c.getPetriNetObjectId())));
-            decryptDataSets(page);
-            page.forEach(this::setImmediateDataFieldsReadOnly);
-        }
-        return page;
+        return repository.findAllBy_idIn(ids).stream()
+                .filter(Objects::nonNull)
+                .sorted(Ordering.explicit(ids).onResultOf(Case::getStringId))
+                .map(caze ->  {
```

Comment:
You can use `peek` instead of `map` in this case, doc: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-

Reply:
`peek` function should be used just for debugging purposes and should not modify elements of the stream

`API Note:
This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline:`